core/lib/zksync_core/src/genesis.rs

```diff
@@ -340,9 +339,12 @@ async fn insert_base_system_contracts_to_factory_deps(
 async fn insert_system_contracts(
     storage: &mut Connection<'_, Core>,
     contracts: &[DeployedContract],
-    chain_id: L2ChainId,
 ) -> Result<(), GenesisError> {
-    let system_context_init_logs = (H256::default(), get_system_context_init_logs(chain_id));
+    let system_context_init_logs = (
+        H256::default(),
+        // During the genesis all chains have the same id.
+        get_system_context_init_logs(L2ChainId::from(DEFAULT_ERA_CHAIN_ID)),
```

Comment:
This will insert an incorrect chain ID for mainnet if someone tries to sync node from scratch. I'm not sure it's currently in the working state but better not to break it even more or at least add a comment

Reply:
Discussed internally. We'll add a comment + create a task